src/models/AbstractWidget.php

```diff
@@ -0,0 +1,126 @@
+<?php namespace Leean\Widgets\Models;
+
+use Leean\Acf\All;
+
+/**
+ * Class AbstractWidget.
+ *
+ * @package Leean\Widgets\Models
+ */
+class AbstractWidget extends \WP_Widget
+{
+	/**
+	 * The slug.
+	 *
+	 * @var bool|string
+	 */
+	private $_slug = '';
+
+	/**
+	 * AbstractWidget constructor.
+	 *
+	 * @param string $title       Widget title
+	 * @param string $description Widget description.
+	 * @param bool   $slug		  Widget slug (use class name if not supplied).
+	 */
+	public function __construct( $title, $description, $slug = false ) {
```

Comment:
I'm not disagreeing with you Cris. It can't be instantiated.
However, it can have a constructor which is extended in the child class (but never called directly).